sklearn/tree/tests/test_monotonic_tree.py

```diff
@@ -79,6 +79,9 @@ def test_monotonic_constraints_classifications(
     est.fit(X_train, y_train)
     y = est.predict_proba(X_test)[:, 1]
 
+    assert np.all(est.predict_proba(X_test) >= 0.0)
```

Comment:
We could also test that the `sum(axis=1)` is close to 1 in terms of proba.

Reply:
Excellent idea !

Reply:
Added in [8ec1ca3](https://github.com/scikit-learn/scikit-learn/pull/27639/commits/8ec1ca3a0b0d7334126645ade911565db8f28c90).